codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/SraOperationOrderTest.java

```diff
@@ -0,0 +1,91 @@
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import software.amazon.smithy.go.codegen.GoCodegenContext;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.knowledge.TopDownIndex;
+
+import static software.amazon.smithy.go.codegen.GoWriter.goTemplate;
+
+/**
+ * Validates that service client operations are performed in the orders specified by the Smithy Reference Architecture (SRA).
+ */
+public class SraOperationOrderTest implements GoIntegration {
+    @Override
+    public void writeAdditionalFiles(GoCodegenContext ctx) {
+        ctx.writerDelegator().useFileWriter("sra_operation_order_test.go", ctx.settings().getModuleName(), writer -> {
+            writer.write(renderCommonTestSource());
+
+            TopDownIndex.of(ctx.model())
+                    .getContainedOperations(ctx.settings().getService(ctx.model()))
+                    .forEach(it -> {
+                        var operationName = ctx.symbolProvider().toSymbol(it).getName();
+                        writer.write(renderTest(operationName));
+                    });
+        });
+    }
+
+    private GoWriter.Writable renderCommonTestSource() {
+        return goTemplate("""
+                $D $D
+                var errTestReturnEarly = errors.New("errTestReturnEarly")
+
+                func captureMiddlewareStack(stack *middleware.Stack) func(*middleware.Stack) error {
+                	return func(inner *middleware.Stack) error {
+                		*stack = *inner
+                		return errTestReturnEarly
+                	}
+                }
+                """, SmithyGoDependency.ERRORS, SmithyGoDependency.SMITHY_MIDDLEWARE);
+    }
+
+    private GoWriter.Writable renderTest(String operationName) {
+        return goTemplate("""
+                $1D $2D $3D $4D $5D $6D
```

Comment:
man I love that we can just import things and then let Go fmt sort things later.